File: Java/7 API ABUSE/1 Reliance on DNS Lookups in a Security Decision/java/MOPAS_1_DNS_lookup.java
/**
 *  MOPAS Reliance on DNS Lookups in a Security Decision Example 1
 */
import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.net.InetAddress;

@WebServlet(urlPatterns = {"/MOPAS_1_DNS_lookup"})
public class MOPAS_1_DNS_lookup extends HttpServlet {

    @Override
    public void doGet(HttpServletRequest request, HttpServletResponse response)
            throws IOException, ServletException {

        boolean trusted = false;
        String ip = request.getRemoteAddr();

        InetAddress addr = InetAddress.getByName(ip);

        if (addr.getCanonicalHostName().endsWith("trustme.com")) {
            trusted = true;
        }

        if (trusted) {
            //do something
        }
    }
}

File: PHP/7 API ABUSE/1 Reliance on DNS Lookups in a Security Decision/php/MOPAS_1_DNS_lookup.php
<?php
/**
 *  MOPAS Reliance on DNS Lookups in a Security Decision Example 1
 */
$hostName = gethostbyaddr($_SERVER['REMOTE_ADDR']);

$trustedHostName = 'partner.trustme.com';
$trusted = FALSE;

$pos = strrpos($hostName, $trustedHostName, -1);

if($pos !== FALSE) {
	$trusted = TRUE;
}

if ($trusted) {
	//do something
}
?>